Source/JavaScriptCore/runtime/VM.cpp

```diff
@@ -1483,10 +1483,15 @@ bool VM::isScratchBuffer(void* ptr)
     return false;
 }
 
+bool VM::hasSyncWaiter()
+{
+    return !!m_syncWaiter;
+}
 Ref<Waiter> VM::syncWaiter()
 {
-    m_syncWaiter->setVM(this);
-    return m_syncWaiter;
+    if (!m_syncWaiter)
+        m_syncWaiter = adoptRef(new Waiter(this));
+    return *m_syncWaiter;
```

Comment:
I think it is fine to always have syncWaiter.

Reply:
Fixed.